prospero-common/src/main/java/org/wildfly/prospero/actions/InstallFolderUtils.java

```diff
@@ -40,6 +41,25 @@ static void verifyIsEmptyDir(Path directory) {
         }
     }
 
+    static Path toRealPath(Path symlink) {
+        Path path = symlink;
+
+        while (path != null && !(Files.exists(path) && Files.isSymbolicLink(path))) {
+            path = path.getParent();
+        }
```

Comment:
To me it looks bit more complex than needed, though I just may be missing something. What is the reason for this cycle?

I'm trying to reproduce the issue, following fails with the FileAlreadyExistsException:

```
Path symbolicLink = Files.createSymbolicLink(Path.of("/tmp/symlink"), Path.of("/tmp"));
Files.createDirectories(Path.of("/tmp/symlink"));
```

But following is OK:

```
Path symbolicLink = Files.createSymbolicLink(Path.of("/tmp/symlink"), Path.of("/tmp"));
Files.createDirectories(Path.of("/tmp/symlink/a")); // deeper paths are OK too
```

So to me it looks like we only need to check the complete path as given.

Reply:
It does seem to work though. I think perhaps it could be simplified but it is fine as it is IMO. :+1:

Reply:
BTW I got the cycle meaning so no need to elaborate :).

What I meant by simplifying is doing just:

```
if (Files.isSymbolicLink(path)) {
    return path.toRealPath();
}
return path;
```

but that's just for consideration...

Reply:
Thanks for review @TomasHofman.

Yes the issue is only reproducible if the top level dir is a symlink, so it would be enough to use the simplified code. 

In this case I wanted to make sure the Path passed further on is consistent regardless if user passed the symlink or a subdirectory of symlink. In my mind it makes it easier to reason about further execution instead of keeping the two possibilities in mind.

If you think it's better to use the simpler approach, I don't mind changing it. Either way I should add this explanation to code :)

Reply:
I think either way is fine, it just took me a bit to decipher that sanitizing method.